### PR TITLE
Adding login by token

### PR DESCRIPTION
**Pull request by k0sm0naft**

### CODE REVIEW

File: src/main/java/ua/bookstore/online/config/SecurityConfig.java [hunk 1]
```diff
@@ -1,45 +1,94 @@
...
+
+    @Bean
+    public AuthenticationManager authenticationManager(
+            AuthenticationConfiguration authenticationConfiguration
+    ) throws Exception {
+        return authenticationConfiguration.getAuthenticationManager();
+    }
+
+    @Bean
+    public OpenAPI customOpenApi() {
```
Comment (yevheniisynytsia) [inline src/main/java/ua/bookstore/online/config/SecurityConfig.java:85]: is this for adding a possibility to login via swagger?

File: src/main/java/ua/bookstore/online/controller/AuthenticationController.java [hunk 2]
```diff
@@ -42,4 +46,18 @@ public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto u
...
     }
+
+    @PostMapping("/login")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @Operation(summary = "Login user", description = "Login user by email and password")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "202", description = "Successfully login"),
+            @ApiResponse(responseCode = "400", description = "Invalid request body",
+                    content = @Content(schema = @Schema(implementation = ErrorResponseDto.class))),
+            @ApiResponse(responseCode = "401", description = "Credentials is wrong",
```
Comment (yevheniisynytsia) [inline src/main/java/ua/bookstore/online/controller/AuthenticationController.java]:
```suggestion
            @ApiResponse(responseCode = "401", description = "Wrong credentials",
```

File: src/main/java/ua/bookstore/online/controller/BookController.java [hunk 3]
```diff
@@ -43,6 +43,8 @@ public class BookController {
             @ApiResponse(responseCode = "201", description = "Successfully created"),
             @ApiResponse(responseCode = "400", description = "Invalid request body",
                     content = @Content(schema = @Schema(implementation = ErrorResponseDto.class))),
+            @ApiResponse(responseCode = "401", description = "Required authorization",
+                    content = @Content(schema = @Schema(implementation = ErrorResponseDto.class))),
             @ApiResponse(responseCode = "403", description = "Not enough access rights",
                     content = @Content(schema = @Schema(implementation = ErrorResponseDto.class))),
```
Comment (yevheniisynytsia) [inline src/main/java/ua/bookstore/online/controller/BookController.java:49]: this really becomes hard to read, but I'm not opposed to it, if you like it

File: src/main/java/ua/bookstore/online/exception/ExceptionHandlerFilter.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.servlet.HandlerExceptionResolver;
+
+@Component
+@RequiredArgsConstructor
+public class ExceptionHandlerFilter extends OncePerRequestFilter {
+    private final HandlerExceptionResolver handlerExceptionResolver;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response,
```
Comment (yevheniisynytsia) [inline src/main/java/ua/bookstore/online/exception/ExceptionHandlerFilter.java:19]: is there a good reason to have it? Isn't `CustomGlobalExceptionHandler` enough?

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]
Looks good, just a couple of comments-questions

File: src/main/java/ua/bookstore/online/config/SecurityConfig.java (hunk 1 above)
Comment (k0sm0naft) [reply src/main/java/ua/bookstore/online/config/SecurityConfig.java:85]: Yes.

File: src/main/java/ua/bookstore/online/exception/ExceptionHandlerFilter.java (hunk 4 above)
Comment (k0sm0naft) [reply src/main/java/ua/bookstore/online/exception/ExceptionHandlerFilter.java:19]: No, `CustomGlobalExceptionHandler` didn't catch exception from filters, they're running before controller.

**Pull request merged**